### PR TITLE
GH-96172 fix unicodedata.east_asian_width being wrong on unassigned code points

### PR DESCRIPTION
**Pull request by cfbolz**
`unicodedata.east_asian_width` returned the wrong value in two situations:
- by default for unassigned code points (should return 'N', did return 'F')
- for reserved but unassigned code points (should return 'W', did return 'F')

This is fixed by changing the default return value through reordering the list `EASTASIANWIDTH_NAMES` and by adding records for unassigned but reserved entries.

* Issue: gh-96172

### CODE REVIEW

File: Tools/unicode/makeunicodedata.py [hunk 1]
```diff
@@ -77,7 +77,7 @@
     "PDF", "EN", "ES", "ET", "AN", "CS", "NSM", "BN", "B", "S", "WS",
     "ON", "LRI", "RLI", "FSI", "PDI" ]
 
-EASTASIANWIDTH_NAMES = [ "F", "H", "W", "Na", "A", "N" ]
+EASTASIANWIDTH_NAMES = [ "N", "H", "W", "Na", "A", "F" ]
```
Comment (isidentical) [inline Tools/unicode/makeunicodedata.py:81]: Would it make sense to clarify it here as well (and also perhaps link the [specification](https://unicode.org/reports/tr11/#Unassigned))?

File: Lib/test/test_unicodedata.py [hunk 2]
```diff
@@ -71,7 +71,7 @@ class UnicodeFunctionsTest(UnicodeDatabaseTest):
 
     # Update this if the database changes. Make sure to do a full rebuild
     # (e.g. 'make distclean && make') to get the correct checksum.
-    expectedchecksum = '98d602e1f69d5c5bb8a5910c40bbbad4e18e8370'
+    expectedchecksum = '4975f3ec0acd4a62465d18c9bf8519b1964181f6'
```
Comment (isidentical) [inline Lib/test/test_unicodedata.py:74]: Would you mind also adding a test that does some direct comparisons (like `assert unicodedata.east_asian_width(X) == Y` for some `Y`'s that used to be wrong).

**Comment by isidentical** [comment]
@cfbolz do you know whether we handle the first case mentioned in "6.1 [Unassigned and Private-Use Characters](https://unicode.org/reports/tr11/#Unassigned)" (Wide for CJK):

```
Unassigned code points in ranges intended for CJK ideographs are classified as Wide. Those ranges are:

the CJK Unified Ideographs block, 4E00..9FFF
the CJK Unified Ideographs Extension A block, 3400..4DBF
the CJK Compatibility Ideographs block, F900..FAFF
the Supplementary Ideographic Plane, 20000..2FFFF
the Tertiary Ideographic Plane, 30000..3FFFF

All other unassigned code points are by default classified as Neutral.
```

**Comment by cfbolz** [comment]
@isidentical oops, I definitely had an explicit test, seems that got lost somehow. will retrieve it tomorrow. and yes, it definitely had characters from the CJK ranges that you list

**Comment by cfbolz** [comment]
it's in my first commit, but then in the third one I removed it again for some reason :-/. https://github.com/python/cpython/pull/96207/commits/5688e6afe15f60d93bc45c275bcd099482ca006b#diff-c31ff7b8fca97de6b4fdaca3e14da27ab3cac411653e9c510a5378b189f909eaR223

fixing tomorrow.

**Comment by isidentical** [comment]
Amazing, thanks!

File: Lib/test/test_unicodedata.py (hunk 2 above)
Comment (cfbolz) [reply Lib/test/test_unicodedata.py:74]: I've added the test back, and also added some cases for characters in private use areas (which need to return "A"), thanks for making me carefully look at the standard again ;-).

**Review by isidentical — APPROVED** [review]

**Pull request merged**